eZ/Publish/Core/FieldType/BinaryBase/Type.php

```diff
@@ -347,20 +347,7 @@ public function validateValidatorConfiguration( $validatorConfiguration )
             switch ( $validatorIdentifier )
             {
                 case 'FileSizeValidator':
-                    if ( !isset( $parameters['maxFileSize'] ) )
-                    {
-                        $validationErrors[] = new ValidationError(
-                            "Validator %validator% expects parameter %parameter% to be set.",
-                            null,
-                            array(
-                                "validator" => $validatorIdentifier,
-                                "parameter" => 'maxFileSize',
-                            ),
-                            "[$validatorIdentifier][maxFileSize]"
-                        );
-                        break;
-                    }
-                    if ( !is_int( $parameters['maxFileSize'] ) && $parameters['maxFileSize'] !== false )
+                    if ( array_key_exists( "maxFileSize", $parameters ) && !is_int( $parameters['maxFileSize'] ) && $parameters['maxFileSize'] !== null )
```

Comment:
@lolautruche That's what I tried to say above - maxFileSize *is* absent, when set to default to null in the type. I tried exactly what you suggest first, but then I removed the if block because it always triggers.